### PR TITLE
Working I2C Devices: BEWARE OF TIMING ISSUES AND K64 RESET CAUSING FATAL DEVICE STATES

### PR DESCRIPTION
**Pull request by spressomartini**
title says it all.

### CODE REVIEW

File: include/drivers/devices/altimu.h [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+int altimu_reg_cmp(uint8_t i2cnum, i2c_seq_t *stat_seq, uint8_t expected, int errcode);
+
+/*
+ * This function initializes the gyroscope and accelerometer of the AltIMU.
+ * Note: This function does not act asynchronously and will block the CPU.
+ * @param i2cnum I2C Channel that is connected to the AltIMU.
+ * @return 0 if no errors, GXL_ERROR_NOT_FOUND if not found, I2C errors if
+ *         transmission fails.
+ */
+int init_altimu_gxl(uint8_t i2cnum);
```
Comment (PyroAVR) [inline include/drivers/devices/altimu.h]: We should make the names here consistent with the peripheral initialization. `altimu_gxl_init`.

File: include/drivers/devices/altimu.h [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+/*
+ * This function reads data from the gyro/accel.
+ * @param i2cnum I2C Channel that is connected to the AltIMU.
+ * @param seq A pointer to the sequence to be sent by the function.
+ * @param status A pointer to a 12 byte buffer to store the status data.
+ * @param done A pointer to a boolean specifying whether or not the transfer is done.
+ * @return 0 if no errors, I2C errors if transmission fails.
+ */
+int read_altimu_gxl(uint8_t i2cnum, uint8_t *data, bool *done);
```
Comment (PyroAVR) [inline include/drivers/devices/altimu.h]: Same with these. The standard is `moduleprefix_function_name`.

File: src/drivers/kinetis/i2c.c [hunk 3]
```diff
@@ -40,7 +40,8 @@ void i2c_init(i2c_config_t config, uint32_t clk_f_hz) {
     base->A1 = 0;
     base->F = 0;
     base->C1 = 0;
-    base->S = 0xFFu;
+    base->S = 0x80;
```
Comment (PyroAVR) [inline src/drivers/kinetis/i2c.c:43]: What change does this make?

**Review by PyroAVR — CHANGES_REQUESTED** [review]
Just a few renaming things.  I can't test this, obviously, since you have the sensor setup.

File: src/drivers/kinetis/i2c.c (hunk 3 above)
Comment (spressomartini) [reply src/drivers/kinetis/i2c.c:43]: apparently `0xFF` was not the reset value of the register and it set a bunch of things that we didn't intend on setting.

File: include/drivers/devices/altimu.h (hunk 1 above)
Comment (spressomartini) [reply include/drivers/devices/altimu.h]: thanks, i keep mixing it up in my head.

**Review by PyroAVR — APPROVED** [review]
full send.

**Comment by PyroAVR** [comment]
squish completed, have fun

**Pull request merged**